Net/src/TCPServerDispatcher.cpp

```diff
@@ -147,16 +147,8 @@ void TCPServerDispatcher::enqueue(const StreamSocket& socket)
 		_queue.enqueueNotification(new TCPConnectionNotification(socket));
 		if (!_queue.hasIdleThreads() && _currentThreads < _pParams->getMaxThreads())
 		{
-			try
-			{
-				_threadPool.startWithPriority(_pParams->getThreadPriority(), *this, threadName);
-				++_currentThreads;
-			}
-			catch (Poco::Exception&)
-			{
-				// no problem here, connection is already queued
-				// and a new thread might be available later.
-			}
+            _threadPool.startWithPriority(_pParams->getThreadPriority(), *this, threadName);
+            ++_currentThreads;
 		}
```

Comment:
It will throw a NoThreadAvailableException. At this point, I'd say we'd want to see all exceptions including that (I don't expect this change to be permanent necessarily). It shouldn't throw that exception in this case since the outer if checks for sufficient threads.